### PR TITLE
feat(common): Add ability to customize Flutter SDK for Workflow pipeline

### PR DESCRIPTION
**Pull request by plasticfiresam**
- Added new input variable for workflow to define exact version of Flutter for another workflow-dependant repos 
- If there is no version specified, will be used previous behavior of using stable branch for flutter checkout

### CODE REVIEW

File: .github/workflows/analysis.yml [hunk 1]
```diff
@@ -1,6 +1,13 @@
...
 
-on: [ workflow_call ]
+on:
+  workflow_call:
+    inputs:
+      # (optional) you can specify which version of Flutter will be used for Analysis workflow, defaults to 3.10.5
+      flutter-version:
+        required: false
+        type: string
+        default: "3.10.5"
```
Comment (maxim-sysoev) [inline .github/workflows/analysis.yml]: может оставить дефолтной версию `stable`, так для ci в принципе ничего не изменится

Comment (plasticfiresam) [reply .github/workflows/analysis.yml]: Убрал значение по умолчанию, и добавил дополнительный шаг с условием, на случай, если не передали версию

**Review by mpkander — APPROVED** [review]
LGTM. Can you test some public lib with new functionality and add results here before merging?

**Comment by plasticfiresam** [comment]
> LGTM. Can you test some public lib with new functionality and add results here before merging?

You can see results of full workflow here:

https://github.com/surfstudio/flutter-otp-autofill/pull/45/checks

TLDR:
image

**Review by maxim-sysoev — APPROVED** [review]
